packages/circuit-ui/components/Checkbox/Checkbox.tsx

```diff
@@ -188,8 +187,9 @@ export const Checkbox = forwardRef(
     }: CheckboxProps,
     ref: CheckboxProps['ref'],
   ) => {
-    const id = customId || uniqueId('checkbox_');
-    const validationHintId = uniqueId('validation_hint-');
+    const id = useId();
+    const checkboxId = customId || id;
```

Comment:
Could this be

```suggestion
    const id = customId || useId();
```

...so we don't generate one if developers pass it?

Reply:
That would violate one of the [rules of hooks](https://legacy.reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level):

> Don’t call Hooks inside loops, conditions, or nested functions.

Reply:
oopsies 😅 I thought I was missing something here